.github/workflows/lance-main.yml

```diff
@@ -0,0 +1,46 @@
+# The lance-main branch is a mirror of the LanceDB main branch, where
+# the rust dependencies use local paths instead of git.
+name: Maintain lance-main branch
+
+on:
+  push:
+    branches:
+      - main
+
+concurrency:
+  # only run one job at a time for this workflow
+  group: ${{ github.workflow }}
+
+jobs:
+  sync:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout commit branch
+        uses: actions/checkout@v3
+        with:
+          path: lancedb
+      - name: Checkout lance main branch
+        uses: actions/checkout@v3
+        with:
+          repository: lancedb/lance
+          path: lance
+      - name: Catch lance-main up to date
+        working-directory: lancedb
+        run: |
+          git checkout lance-main
+          git fetch origin main
+          for commit in $(git rev-list lance-main..main --reverse); do
+            git cherry-pick $commit --strategy-option=theirs
+            python ci/set_lance_version.py local
+            git add Cargo.toml Cargo.lock
+            git commit -m "use local lance" --allow-empty --no-verify
+          done
```

Comment:
Is this for loop expected to only run once?  Also, won't this create a bunch of `use local lance` commits? (not sure if it matters)

Could you just do `git reset --hard origin/main` instead of the for loop (though it would have to be before you checked out lance?